sbt/src/sbt-test/project/sbt-composite-projects/changes/shadow.sbt

```diff
@@ -0,0 +1,32 @@
+import sbt.internal.AddSettings
+import sbt.CompositeProject
+
+lazy val check = taskKey[Unit]("check")
+
+// Based on sbt-file-projects test
+lazy val foo = new CompositeProject
+{
+  val jvm = Project.apply("jvm", new File("jvm"))
+  val js = Project.apply("js", new File("js"))
+  def componentProjects: Seq[Project] = Seq(jvm, js)
+}
+
+lazy val fooJVM = foo.jvm.settings(version := "0.2.0") // this one needs to win
```

Comment:
This is very good.

To make it even better, you can also declare the `lazy val fooJS`, but *above* the `lazy val foo` (they're lazy vals, so it will work). But that should ensure that the shadowing treatment does not accidentally work *via* the ordering (which I suspect is currently the case).

Reply:
>  (which I suspect is currently the case).

Indeed, adding a `lazy val fooJS` first does fail...